File: v1.1/rest_delete.pl
#!/usr/bin/perl -w
use strict;
use warnings;
# Create a user agent object
use LWP::UserAgent;
my $ua = LWP::UserAgent->new;
$ua->agent("MyApp/0.1");

# Create a request
my $req = HTTP::Request->new(DELETE => 'http://vengersonline.com:5000/api/nodes/192.168.1.100');
$req->content_type('application/json');
$req->authorization_basic("pi-dome", "pi-dome");

# Pass request to the user agent and get a response back
my $res = $ua->request($req);

# Check the outcome of the response
if ($res->is_success) {
  print $res->content;
} else {
  print $res->status_line, "n";
}

File: v1.1/rest_delete.py
#!/usr/bin/python
import requests
import httplib
import base64
import string
import json


#requests.delete(url)

#requests.delete("vengersonline.com:5000/api/nodes/192.168.1.100")
#delete_url = "vengersonline.com:5000/api/nodes/192.168.1.100"

#req = urllib2.Request(delete_url, None, headers)<br>req.get_method = lambda: 'DELETE'
#data = urllib2.urlopen(req).read()
#result = json.loads(data)


#def _make_request(url, data, method):
#    request.urllib2.Request(url, data=data)
#    request.get_method = lambda: method
#url = "http://vengersonline.com:5000/api/nodes/192.168.1.100"
#url = "http://vengersonline.com:5000/api/nodes/192.168.1.100"
#requests.delete(url)



import urllib2
#opener = urllib2.build_opener(urllib2.HTTPHandler)
#request = urllib2.Request('http://vengersonline.com:5000/api/nodes/192.168.1.100')
#request.add_header('Content-Type', 'your/contenttype')
#request.get_method = lambda: 'DELETE'
#url = opener.open(request)

def delete_node(id):
   query_url = "http://vengersonline.com:5000/api/nodes/" + id
   #query_url = oauth['http://vengersonline.com:5000']+'/api/nodes/%s' % id 
   #headers = {
   #  'Authorization': 'OAuth '+oauth['access_token']
   #}
   opener = urllib2.build_opener(urllib2.HTTPHandler)
   req = urllib2.Request(query_url, None)
   #req = urllib2.Request(query_url, None, headers)
   req.get_method = lambda: 'DELETE' 
   url = urllib2.urlopen(req) 

delete_node("192.168.1.100")

